Sources/SpeziViews/Model/ViewState.swift

```diff
@@ -10,6 +10,52 @@ import Foundation
 
 
 /// Allows SwiftUI views to keep track of their state and communicate to outside views.
+///
+/// A `ViewState` provides a built-in mechanism for tracking the state of a Spezi UI component.
+/// A view can be in either an `idle`, `processing`, or `error` `ViewState`.
+///
+/// ```swift
+/// import SpeziViews
+/// import SwiftUI
+///
+/// enum ProcessingError: LocalizedError {
+///     case resourceError
+/// }
+///
+/// func doSomeProcessing() throws {
+///     // Function may throw an error
+///     throw ProcessingError.resourceError
+/// }
```

Comment:
I would suggest to remove these types here and just add a `// ...` placeholder in the button definition that entails that it could call a throwing and async function in here.